src/main/scala/beam/router/BeamSkimmer.scala

```diff
@@ -474,6 +495,99 @@ class BeamSkimmer @Inject()(val beamConfig: BeamConfig, val beamServices: BeamSe
     }
     writer.close()
   }
+
+  // **********
+  // Skim Plus
+  private var previousSkimsPlus: TrieMap[(TimeBin, Id[TAZ], Id[VehicleManager], Label), Double] =
+    initialPreviousSkimsPlus()
+  private var skimsPlus: TrieMap[(TimeBin, Id[TAZ], Id[VehicleManager], Label), Double] = TrieMap()
+  private val timeWindow: Int = 5 * 60
+  private def skimsPlusFilePath: Option[String] = {
+    val maxHour = TimeUnit.SECONDS.toHours(new TravelTimeCalculatorConfigGroup().getMaxTime).toInt
+    BeamWarmStart(beamConfig, maxHour).getWarmStartFilePath(observedSkimsPlusFileBaseName + ".csv.gz")
+  }
+
+  def getPreviousSkimPlusValues(
+    startTime: Int,
+    endTime: Int,
+    taz: Id[TAZ],
+    vehicleManager: Id[VehicleManager],
+    label: Label
+  ): Vector[Double] = {
+    (timeToBin(startTime, timeWindow) to timeToBin(endTime, timeWindow))
+      .map { i =>
+        previousSkimsPlus.get((i, taz, vehicleManager, label))
+      }
+      .toVector
+      .flatten
+  }
+
+  def observeVehicleDemandByTAZ(
+    time: Int,
+    location: Coord,
+    vehicleManager: Id[VehicleManager],
+    label: Label,
+    who: Option[Id[Person]]
+  ): Unit = {
+    who match {
```

Comment:
If you just need to do some side-effect on `who` in case it is `Some`, you could use `foreach`:
```
who.foreach { _ =>
val timeBin = timeToBin(time, timeWindow)
        val taz = beamServices.tazTreeMap.getTAZ(location.getX, location.getY)
        val key = (timeBin, taz.tazId, vehicleManager, label)
        skimsPlus.put(key, skimsPlus.getOrElse(key, 1.0) + 1.0)
}
```

Reply:
actually, I already removed the who from the parameters. I realized afterwards that I don't need it.